packages/core/src/queries/passcode.ts

```diff
@@ -0,0 +1,36 @@
+import { PasscodeType, Passcode, Passcodes, CreatePasscode } from '@logto/schemas';
+import { sql } from 'slonik';
+
+import { buildInsertInto } from '@/database/insert-into';
+import pool from '@/database/pool';
+import { buildUpdateWhere } from '@/database/update-where';
+import { convertToIdentifiers } from '@/database/utils';
+import { DeletionError } from '@/errors/SlonikError';
+
+const { table, fields } = convertToIdentifiers(Passcodes);
+
+export const findUnconsumedPasscodeBySessionIdAndType = async (
```

Comment:
`.one()` throws an error if not found, also we should consider multiple unconsumed passcodes? maybe `.many()`?

Reply:
if so, we can delete those passcodes in batch (use pg `in` or `=any()` clause to test array)

Reply:
The number of unconsumed passcode for specific type and session is desigend to be 1 or 0.

But treat it as "multiple" would be more safer, updated.